pkgs/development/libraries/qt-6/modules/qtdeclarative.nix

```diff
@@ -15,12 +15,22 @@ qtModule {
   strictDeps = true;
 
   patches = [
-    # prevent headaches from stale qmlcache data
-    ../patches/0001-qtdeclarative-disable-qml-disk-cache.patch
+    # invalidates qml caches created with different qtdeclarative derivations
+    ../patches/qtdeclarative-use-nix-qml-compile-tag.patch
+    # includes store paths of applications in qml cache hashes
+    ../patches/qtdeclarative-include-store-paths-in-qmlc-paths.patch
     # add version specific QML import path
     ../patches/0002-qtdeclarative-also-use-versioned-qml-paths.patch
   ];
 
+  # Only qml files with a matching tag will be loaded by qtdeclarative,
+  # so it is set to the unique store path of the build.
+  # The maximum length of the tag is 48 bytes, so the path is trimmed to
```

Comment:
Should we maybe also trim /nix/store/ and leave more from the end?

Reply:
I got the whole hash already. Are hash conflicts common enough that this should be a concern?

Reply:
the hashes of the nix store entries should never collide.